### DISCUSSION TITLE
brew upgrade fails. brew uninstall fails. cannot uninstall anything.

### DISCUSSION THREAD

**Discussion by DaBigBlob** [Homebrew/discussions#4181] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20-309-g206d4c8
ORIGIN: https://github.com/Homebrew/brew
HEAD: 206d4c86b36d2d9e093dc886988be9fbb2b28a12
Last commit: 43 minutes ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2cdd0039cb43372f0334b316920024bca3ea4f05
Core tap last commit: 4 days ago
Core tap branch: master
Core tap JSON: 07 Feb 14:19 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  ufraw
```

### Description of issue

brew upgrade fails
```txt
---START TERMINAL---
$ brew update && brew upgrade

Already up-to-date.
Error: No file found at https://raw.githubusercontent.com/Homebrew/homebrew-cask/9da035cce5f43a7a91305908c919cda2d1643571/Casks/blender.rb
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/opt/homebrew/Library/Homebrew/api.rb:91:in `fetch_file_source'
/opt/homebrew/Library/Homebrew/api/cask.rb:20:in `fetch_source'
/opt/homebrew/Library/Homebrew/cask/cask_loader.rb:242:in `load'
/opt/homebrew/Library/Homebrew/cask/cask_loader.rb:382:in `load'
/opt/homebrew/Library/Homebrew/cask/caskroom.rb:50:in `block in casks'
/opt/homebrew/Library/Homebrew/cask/caskroom.rb:46:in `map'
/opt/homebrew/Library/Homebrew/cask/caskroom.rb:46:in `casks'
/opt/homebrew/Library/Homebrew/cask/cmd/upgrade.rb:98:in `upgrade_casks'
/opt/homebrew/Library/Homebrew/cmd/upgrade.rb:226:in `upgrade_outdated_casks'
/opt/homebrew/Library/Homebrew/cmd/upgrade.rb:117:in `upgrade'
/opt/homebrew/Library/Homebrew/brew.rb:93:in `<main>'
---END TERMINAL---
```
<img width="1388" alt="Screenshot 2023-02-07 at 8 49 28 PM" src="https://user-images.githubusercontent.com/73036332/217285850-8048c448-37ad-44af-a286-4fc0887a750a.png">

**Comment by gromgit** [comment]
`export HOMEBREW_NO_INSTALL_FROM_API=1` and try your `brew` commands again.

**Reply by gcr** [reply]
this workaround temporarily fixes my issue for that command!

however, future commands fail unless I set that environment variable, even after a `brew update`. Does anyone know of a permanent fix?

**Comment by gcr** [comment]
I'm also affected by this exact issue. In my case, brew fails to find `https://raw.githubusercontent.com/Homebrew/homebrew-cask/9da035cce5f43a7a91305908c919cda2d1643571/Casks/blender.rb`, then after two `brew update`s and a `brew doctor`, it then fails to find `https://raw.githubusercontent.com/Homebrew/homebrew-cask/9da035cce5f43a7a91305908c919cda2d1643571/Casks/blackhole-2ch.rb`

**Reply by gcr** [reply]
now without any changes, it fails to find `https://raw.githubusercontent.com/Homebrew/homebrew-cask/9da035cce5f43a7a91305908c919cda2d1643571/Casks/inkscape.rb`. If I load them in my browser, the above two links fail, but this one works.

I wonder if we're in the middle of a bad deploy?

**Comment by changhoon-sung** [comment]
Phew.... I'm not the only one. Same here. I think it is something kind of brew or tap problem as gcr said.
I'm even failing in `brew bundle dump` for reset the brew. It is bad..

**Comment by timothyarthur** [comment]
I'm also having this problem. The error seems to alternate between a few different casks not found when I run `brew upgrade`. For me, the error has thrown on `https://raw.githubusercontent.com/Homebrew/homebrew-cask/9da035cce5f43a7a91305908c919cda2d1643571/Casks/anaconda.rb`,`https://raw.githubusercontent.com/Homebrew/homebrew-cask/9da035cce5f43a7a91305908c919cda2d1643571/Casks/firefox.rb` and `https://raw.githubusercontent.com/Homebrew/homebrew-cask/9da035cce5f43a7a91305908c919cda2d1643571/Casks/vmware-fusion.rb` so far.

**Reply by timothyarthur** [reply]
I'm getting intermittent 404s when I navigate to these links by web browser. Is this a problem with github's servers?

**Comment by Rylan12** [comment]
There is an issue on GitHub's end with how we were distributing formulae and casks. We've temporarily rolled that feature back until GitHub fixes the issue. Run `brew update` and then things should start to work for you again 🤞

**Reply by mikesep** [reply]
Things are working for me again today. Thanks @Rylan12!

**Reply by lm782** [reply]
Similar sort of problem. brew stopped working for me as well (MacOS 12.6.3 using zsh). I tried uninstalling and reinstalling brew and still have problems. 

"Luis@Luis-Mini-3 ~ % brew update
error: unable to create file Formula/findent.rb: No such file or directory
error: unable to create file Formula/pypy.rb: No such file or directory
error: unable to create file Formula/zint.rb: No such file or directory
fatal: Could not reset index file to revision 'HEAD'.
Error: Could not 'git stash' in /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core!
Please stash/commit manually if you need to keep your changes or, if not, run:
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
  git reset --hard origin/master
Luis@Luis-Mini-3 ~ % cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
Luis@Luis-Mini-3 homebrew-core % git reset --hard origin/master
error: unable to create file Formula/findent.rb: No such file or directory
error: unable to create file Formula/http-server.rb: No such file or directory
error: unable to create file Formula/pypy.rb: No such file or directory
error: unable to create file Formula/tlx.rb: No such file or directory
error: unable to create file Formula/zint.rb: No such file or directory
Updating files: 100% (6887/6887), done.
fatal: Could not reset index file to revision 'origin/master'.
Luis@Luis-Mini-3 homebrew-core % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   D .github/ISSUE_TEMPLATE/bug.md
   M .github/ISSUE_TEMPLATE/config.yml
   M .github/PULL_REQUEST_TEMPLATE.md
   D .github/stale.yml
   D .github/support.yml
   M .github/workflows/publish-commit-bottles.yml
   M .github/workflows/scheduled.yml
   M .github/workflows/tests.yml
   D Aliases/ansible@2.9
   D Aliases/autoconf@2.69
   D Aliases/bison@3.6
   D Aliases/boost-python@1.72..." and lists many more files. 
"Luis@Luis-Mini-3 homebrew-core % cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f
Saved working directory and index state WIP on master: c40b0f25552 sundials: update 5.3.0 bottle.
warning: failed to remove Formula/fava.rb: No such file or directory
warning: failed to remove Formula/openfpgaloader.rb: No such file or directory
warning: failed to remove Formula/python-gdbm@3.11.rb: No such file or directory
warning: failed to remove Formula/simdutf.rb: No such file or directory"

Not sure what to do next.

**Reply by gromgit** [reply]
@lm782 Your Homebrew setup seems to be messed up. Try `brew update-reset`, then `brew doctor` to see if that fixes things.

**Reply by lm782** [reply]
Details for reset, doctor and stash below. 

Luis@Luis-Mini-3 ~ % brew update-reset
==> Fetching /usr/local/Homebrew...
remote: Enumerating objects: 468, done.
remote: Counting objects: 100% (227/227), done.
remote: Total 468 (delta 227), reused 227 (delta 227), pack-reused 241
Receiving objects: 100% (468/468), 180.18 KiB | 4.00 MiB/s, done.
Resolving deltas: 100% (340/340), completed with 131 local objects.
From https://github.com/Homebrew/brew
 * [new branch]          imgbot     -> origin/imgbot
   b0a74e532..ed7938115  master     -> origin/master

==> Resetting /usr/local/Homebrew...
branch 'master' set up to track 'origin/master'.
Switched to and reset branch 'master'
Your branch is up to date with 'origin/master'.

==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask...
remote: Enumerating objects: 2178, done.
remote: Counting objects: 100% (1609/1609), done.
remote: Compressing objects: 100% (23/23), done.
remote: Total 2178 (delta 1593), reused 1601 (delta 1586), pack-reused 569
Receiving objects: 100% (2178/2178), 683.16 KiB | 9.11 MiB/s, done.
Resolving deltas: 100% (1595/1595), completed with 495 local objects.
From https://github.com/Homebrew/homebrew-cask
   b69d2bf867..8784f1a42f  master     -> origin/master

==> Resetting /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask...
branch 'master' set up to track 'origin/master'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
remote: Enumerating objects: 3464, done.
remote: Counting objects: 100% (2657/2657), done.
remote: Compressing objects: 100% (162/162), done.
remote: Total 3464 (delta 2503), reused 2644 (delta 2495), pack-reused 807
Receiving objects: 100% (3464/3464), 1.97 MiB | 4.30 MiB/s, done.
Resolving deltas: 100% (2523/2523), completed with 391 local objects.
From https://github.com/Homebrew/homebrew-core
   c26f1c8714c..6bcb7a89179  master     -> origin/master

==> Resetting /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
error: unable to create file Formula/fcrackzip.rb: No such file or directory
error: unable to create file Formula/findent.rb: No such file or directory
error: unable to create file Formula/http-server.rb: No such file or directory
error: unable to create file Formula/jose.rb: No such file or directory
error: unable to create file Formula/lammps.rb: No such file or directory
error: unable to create file Formula/pypy.rb: No such file or directory
error: unable to create file Formula/terraforming.rb: No such file or directory
error: unable to create file Formula/texi2html.rb: No such file or directory
error: unable to create file Formula/tlx.rb: No such file or directory
error: unable to create file Formula/zint.rb: No such file or directory
Updating files: 100% (6901/6901), done.
branch 'master' set up to track 'origin/master'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

Luis@Luis-Mini-3 ~ % brew doctor
==> Downloading https://formulae.brew.sh/api/formula.json
######################################################################## 100.0%
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   D Formula/fcrackzip.rb
   D Formula/findent.rb
   D Formula/http-server.rb
   D Formula/jose.rb
   D Formula/lammps.rb
   D Formula/pypy.rb
   D Formula/terraforming.rb
   D Formula/texi2html.rb
   D Formula/tlx.rb
   D Formula/zint.rb
Luis@Luis-Mini-3 ~ %  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f
Saved working directory and index state WIP on master: 6bcb7a89179 esbuild: update 0.17.8 bottle.
error: unable to create file Formula/fcrackzip.rb: No such file or directory
error: unable to create file Formula/findent.rb: No such file or directory
error: unable to create file Formula/http-server.rb: No such file or directory
error: unable to create file Formula/jose.rb: No such file or directory
error: unable to create file Formula/lammps.rb: No such file or directory
error: unable to create file Formula/pypy.rb: No such file or directory
error: unable to create file Formula/terraforming.rb: No such file or directory
error: unable to create file Formula/texi2html.rb: No such file or directory
error: unable to create file Formula/tlx.rb: No such file or directory
error: unable to create file Formula/zint.rb: No such file or directory
fatal: Could not reset index file to revision 'HEAD'.
Luis@Luis-Mini-3 homebrew-core % 

Any concerns about those errors?

**Reply by gromgit** [reply]
Yes. This:
```
Uncommitted files:
D Formula/fcrackzip.rb
D Formula/findent.rb
D Formula/http-server.rb
D Formula/jose.rb
D Formula/lammps.rb
D Formula/pypy.rb
D Formula/terraforming.rb
D Formula/texi2html.rb
D Formula/tlx.rb
D Formula/zint.rb
```
says the named files have been deleted in your filesystem, when they should exist. What's the output of `brew config`?

**Comment by lm782** [comment]
Luis@Luis-Mini-3 homebrew-core % brew config
HOMEBREW_VERSION: 3.6.21-89-ged79381
ORIGIN: https://github.com/Homebrew/brew
HEAD: ed793811553eb6fca3dd602931e37b7187f86a68
Last commit: 5 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6bcb7a891795ba378edd7b70afa89cf350a327c9
Core tap last commit: 4 days ago
Core tap branch: master
Core tap JSON: 13 Feb 14:09 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.3-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Luis@Luis-Mini-3 homebrew-core %

**Reply by gromgit** [reply]
Time to wipe and redownload the core repo:
```
rm -fr $(brew --repo homebrew/core)
brew tap homebrew/core
```
then run `brew doctor` again.

**Comment by lm782** [comment]
Permission denied. Any concerns using sudo? 

Luis@Luis-Mini-3 homebrew-core % rm -fr $(brew --repo homebrew/core)
rm: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula: Permission denied
rm: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core: Permission denied

**Reply by gromgit** [reply]
Did you install Homebrew as a different user? What's the output of:
```
ls -la $(brew --repo homebrew/core)
```
?

**Comment by lm782** [comment]
Homebrew was installed so long ago I don't recall details (e.g. if I was prompted for sudo privileges). Considering the 'ls -la' output below includes 'admin', looks like I was (prompted for privileges). Most all other files list 'staff' in that column. 

Luis@Luis-Mini-3 homebrew-core % ls -la $(brew --repo homebrew/core)
total 0
drwxr-xr-x     3 Luis  admin     96 Feb 17 11:11 .
drwxr-xr-x     4 Luis  admin    128 Dec 14 21:29 ..
drwxr-xr-x  1673 Luis  admin  53536 Feb 17 11:11 Formula
Luis@Luis-Mini-3 homebrew-core %

**Reply by gromgit** [reply]
If you're logged in as `Luis`, then you should be able to remove the entire `homebrew/core` given the above permissions, so there's something extra happening:
1. What's the output of `ls -lae@ $(brew --repo homebrew/core)`?
2. Are you running any anti-virus or other third-party system monitoring/management software?

**Comment by lm782** [comment]
Sorry for the slow response. 
Luis@Luis-Mini-3 ~ % ls -lae@ $(brew --repo homebrew/core)
total 0
drwxr-xr-x     3 Luis  admin     96 Feb 17 11:11 .
drwxr-xr-x     4 Luis  admin    128 Dec 14 21:29 ..
drwxr-xr-x  1673 Luis  admin  53536 Feb 17 11:11 Formula
Luis@Luis-Mini-3 ~ % 

My previous employer loaded an enterprise management package which appears to be still installed and running. One of the components is anti-virus software (CrowdStrike AV).

**Reply by gromgit** [reply]
Can you try temporarily turning off the AV software, then `rm -fr $(brew --repo homebrew/core)`?

**Reply by lm782** [reply]
I'm not familiar with CrowdStrike. The webpages I found say, to verify if it is installed, you look for Falcon.app. Falcon is not on my machine. I tried again and "rm -fr $(brew --repo homebrew/core)" still gives the same Permission Denied error. 

The /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula directory is not empty (7 files). Should I delete the 7 files manually then remove the Formula directory, and then the homebrew-core directory?

**Reply by gromgit** [reply]
If `rm -fr` is only complaining about the two directories, I'm surprised that there are still files therein. What's the output of `ls -lae@R $(brew --repo homebrew/core)`?

**Reply by lm782** [reply]
Luis@Luis-Mini-3 ~ % ls -lae@R $(brew --repo homebrew/core)
total 0
drwxr-xr-x     3 Luis  admin     96 Feb 17 11:11 .
drwxr-xr-x     4 Luis  admin    128 Dec 14 21:29 ..
drwxr-xr-x  1666 Luis  admin  53312 Feb 21 21:12 Formula

/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula:
total 56
drwxr-xr-x  1666 Luis  admin  53312 Feb 21 21:12 .
drwxr-xr-x     3 Luis  admin     96 Feb 17 11:11 ..
-rw-r--r--     1 Luis  admin   1490 Feb  8 11:59 daemon.rb
-rw-r--r--     1 Luis  admin   2135 Feb  8 13:58 jose.rb
-rw-r--r--     1 Luis  admin   3461 Feb  8 13:58 lammps.rb
-rw-r--r--     1 Luis  admin   5447 Feb  8 13:58 terraforming.rb
-rw-r--r--     1 Luis  admin   2142 Feb  8 13:58 texi2html.rb
-rw-r--r--     1 Luis  admin   2957 Feb  8 13:58 tmx.rb
Luis@Luis-Mini-3 ~ %

**Reply by gromgit** [reply]
What happens when you "rm -f $(brew --repo homebrew/core)/Formula/*"?

If that still gets you errors, then try turning off SIP and try the `rm` again: https://www.imore.com/how-turn-system-integrity-protection-macos. (This shouldn't be needed, since the `/usr/local` hierarchy is specifically _not_ covered by SIP, but I'm running out of ideas.)

If that still doesn't work, and given this:
>My previous employer loaded an enterprise management package which appears to be still installed and running. One of the components is anti-virus software (CrowdStrike AV).
>...
>I'm not familiar with CrowdStrike. The webpages I found say, to verify if it is installed, you look for Falcon.app. Falcon is not on my machine.

something you don't know about it is running on your system, and preventing you from doing stuff that should be allowed. You may want to back up your system, do a fresh macOS install, then restore files _as necessary_. Think of it as a good time to spring-clean your Mac.

**Reply by lm782** [reply]
I didn't get an error and it seems to have worked, but "ls" shows there's at least one file (tmx.rb) remaining. 

Luis@Luis-Mini-3 ~ % rm -f $(brew --repo homebrew/core)/Formula/*                                                                           
zsh: sure you want to delete all 7 files in /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula [yn]? y
Luis@Luis-Mini-3 ~ % 
Luis@Luis-Mini-3 ~ % ls -lae@R $(brew --repo homebrew/core)      
total 0
drwxr-xr-x     3 Luis  admin     96 Feb 17 11:11 .
drwxr-xr-x     4 Luis  admin    128 Dec 14 21:29 ..
drwxr-xr-x  1627 Luis  admin  52064 Feb 23 10:40 Formula

/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula:
total 8
drwxr-xr-x  1627 Luis  admin  52064 Feb 23 10:40 .
drwxr-xr-x     3 Luis  admin     96 Feb 17 11:11 ..
-rw-r--r--     1 Luis  admin   2957 Feb  8 13:58 tmx.rb
Luis@Luis-Mini-3 ~ %

**Reply by gromgit** [reply]
What's the output of `lsof $(brew --repo homebrew/core)/Formula/*`?

**Reply by lm782** [reply]
Luis@Luis-Mini-3 homebrew-core % lsof $(brew --repo homebrew/core)/Formula/*
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/jose.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libxls.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/terraforming.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/tmx.rb: No such file or directory
lsof 4.91
 latest revision: ftp://lsof.itap.purdue.edu/pub/tools/unix/lsof/
 latest FAQ: ftp://lsof.itap.purdue.edu/pub/tools/unix/lsof/FAQ
 latest man page: ftp://lsof.itap.purdue.edu/pub/tools/unix/lsof/lsof_man
 usage: [-?abhlnNoOPRtUvVX] [+|-c c] [+|-d s] [+D D] [+|-f[cgG]]
 [-F [f]] [-g [s]] [-i [i]] [+|-L [l]] [+|-M] [-o [o]] [-p s]
 [+|-r [t]] [-s [p:s]] [-S [t]] [-T [t]] [-u s] [+|-w] [-x [fl]] [--] [names]
Use the ``-h'' option to get more help information.
Luis@Luis-Mini-3 homebrew-core %

**Reply by gromgit** [reply]
Now things are getting VERY weird: You have formulae appearing and disappearing almost at random. That's definitely NOT normal.

I suggest you wipe and reinstall your Mac at this point.

**Reply by lm782** [reply]
Agreed, that is strange behavior. I started looking into MacOS and malware to get better informed. Rebuilding this machine from scratch would be a BIG job. Will start with a bit more troubleshooting before pulling that trigger. 

First step is a reboot, which seems to have done something or another because the system's behavior is now a little different. Looks better to me. Is this what you were expecting? 

Luis@Luis-Mini-3 ~ % rm -f $(brew --repo homebrew/core)/Formula/*
zsh: sure you want to delete all 4 files in /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula [yn]? y
Luis@Luis-Mini-3 ~ % lsof $(brew --repo homebrew/core)/Formula/* 
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/clockywock.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/clutter-gst.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cminpack.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/crash.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/csmith.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/dcfldd.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/detekt.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fbi-servefiles.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fcrackzip.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fifechan.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/findent.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/firebase-cli.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fits.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/flawfinder.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fortio.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fuse-emulator.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gwyddion.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/hcxtools.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/http-server.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/innotop.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/inspircd.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/instalooter.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/instead.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/intercal.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/intltool.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ios-deploy.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/iso-codes.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/kibana.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/lammps.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libxls.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libzdb.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mergepbx.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/micro.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/most.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pachi.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pypy.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/redis-leveldb.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/rp.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/salt.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/scummvm-tools.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/signal-cli.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/slashem.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/texi2html.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/tlx.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xalan-c.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xmlcatmgr.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xmltoman.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/yubico-piv-tool.rb: No such file or directory
lsof: status error on /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/zint.rb: No such file or directory
Luis@Luis-Mini-3 ~ %

**Reply by gromgit** [reply]
1. Your shell is expanding `$(brew --repo homebrew/core)/Formula/*` into a continually changing number of formulae (4 for the first command, at least 49 for the second).
2. The files your shell is seeing...no longer exist when any command you run tries to do something with/to them. (I'd bet `rm $(brew --repo homebrew/core)/Formula/*`, i.e. without the `-f` flag, would also throw "No such file or directory" errors.)

There's nothing "better" about any of that. I can't even predict what you'll see if you `ls -l $(brew --repo homebrew/core)/Formula/`: you might see no files, 4 files, 49 files, thousands of files, or something else.

**Reply by lm782** [reply]
Ugh, got it. Thanks. Will look into building anew...
